### PR TITLE
Add auth feature

### PR DESCRIPTION
**Pull request by agrawalshanu708**
Add authentication
users can register through signup form
and can login

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -10,7 +10,25 @@ display: flex;
...
+  min-width: 10%;
+  height:3rem;
+  background-color:transparent;
+  margin: 1rem 0;
+  padding:1rem;
+  display: flex;
+  align-items: center;
+} 
+.primary_btn{
+  background-color:#090a0c;
```
Comment (MonikaJhingan) [inline src/App.css:23]: Use var in place of direct color codes.

**Review by therajatg — APPROVED** [review]
code looks good

**Review by k-rahul21 — APPROVED** [review]

File: src/App.css [hunk 2]
```diff
@@ -10,7 +10,25 @@ display: flex;
...
+  color: #f3f2f1;
+  border: none;
+}
+.button:hover{
+  border: 1px solid #090a0c;
+  color: #090a0c;
```
Comment (MonikaJhingan) [inline src/App.css:29]: Same here.

File: src/pages/auth/signup/signup.jsx [hunk 3]
```diff
@@ -1,63 +1,119 @@
...
+  </div> */}
 
-  <div class="grid-item">
-  <label htmlFor="user-pass">Create Password</label>
-  <input id="user-pass" type="password" />
-  </div>
+              <div class="grid-item">
+                <label htmlFor="user-pass">Create Password</label>
+                <input
+                  id="user-pass"
```
Comment (MonikaJhingan) [inline src/pages/auth/signup/signup.jsx:87]: The code is well implemented and use of useState has been done nicely and all modules are used perfectly just a small syggestion regarfing indentation of the code.

File: src/pages/auth/signup/signup.jsx [hunk 4]
```diff
@@ -1,63 +1,119 @@
...
-  <input id="user-signup-email" type="email" />
-  </div>  
+              <div class="grid-item">
+                <label htmlFor="user-signup-email">Email</label>
+                <input
+                  id="user-signup-email"
+                  type="email"
+                  onChange={(e) => {
+                    e.preventDefault()
+                    setEmail(e.target.value)}}
```
Comment (MonikaJhingan) [inline src/pages/auth/signup/signup.jsx:75]: Target value has been put correctly.

File: src/pages/auth/login/login.jsx [hunk 5]
```diff
@@ -1,67 +1,122 @@
...
+ const body = {
+email : email,
+password : password,
+ }
+
+ try {
+     const response = await axios.post("/api/auth/login", body)
+     response.data.encodedToken? navigate("/"):alert("login failed")
+ } catch (error) {
+    alert("Error")
```
Comment (MonikaJhingan) [inline src/pages/auth/login/login.jsx:26]: Handling of error using throw catch concept  is good.

**Review by MonikaJhingan — APPROVED** [review]

**Pull request merged**